crates/bin/cli-wallet/src/main.rs

```diff
@@ -241,7 +246,26 @@ async fn main() -> Result<()> {
             let wad = opt_proofs
                 .map(|proofs| Wad { node_url, proofs })
                 .ok_or(anyhow!("Not enough funds"))?;
-            println!("Wad:\n{}", serde_json::to_string(&wad)?);
+
+            match output {
+                Some(output_path) => {
+                    if let Some(ext) = output_path.extension() {
+                        if ext != "json" {
+                            return Err(anyhow!("Error: Output file must have a .json extension."));
+                        }
+                    } else {
+                        return Err(anyhow!("Output file should be a `.json` file"));
+                    }
```

Comment:
We don't want to fail for an invalid extension issue after we did all the work of processing the tokens in db.

So, let's do those checks at the very beginning of the execution.
Something like this, at the very beginning.
```rust
let output: Option<PathBuff> = output.map(|output_path| 
  if output_path.extension().ok_or_esle(anyhow!("Error: Output file must have a .json extension."))? == "json" {
   Ok(output_path)
  } else { Err(anyhow!("Output file should be a `.json` file")) }
).transpose()?;
```